### PR TITLE
On successful registration, users should receive a JWT

### PR DESCRIPTION
**Pull request by AimePazzo**
image

### CODE REVIEW

File: dist/src/helpers/index.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+"use strict";
+var __importDefault = (this && this.__importDefault) || function (mod) {
+    return (mod && mod.__esModule) ? mod : { "default": mod };
+};
+Object.defineProperty(exports, "__esModule", { value: true });
+exports.generateToken = void 0;
+// Helpers
+const jsonwebtoken_1 = __importDefault(require("jsonwebtoken"));
+const generateToken = (user) => {
```
[the remark this answers is not in the document]
Comment (AimePazzo) [reply dist/src/helpers/index.js]: fixed

File: src/modules/auth/validation/authValidations.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    gender: "male" | "female" | "other";
+    birthDate: string;
+    language: string;
+    currency: string;
+    role: "buyer" | "seller" | "admin";
+}
+
+const authSchema = Joi.object<User>({
+    firstName: Joi.string().required().messages({
+        "string.base": "First name should be a type of text",
```
Comment (key-joshua) [inline src/modules/auth/validation/authValidations.ts:19]: make sure to return the exact attribute name into the message. 
i.e, If attribute name is ```firstName```. The message should be "firstName is required"

```
firstName: Joi.string().required().messages({
        "string.base": "firstName should be a type of text",
        "string.empty": "firstName cannot be an empty field",
        "any.required": "firstName is required"
    }),
```

File: src/modules/auth/validation/authValidations.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        "string.empty": "Last name cannot be an empty field",
+        "any.required": "Last name is required"
+    }),
+    email: Joi.string().email().required().messages({
+        "string.base": "Email should be a type of text",
+        "string.email": "Email must be a valid email",
+        "string.empty": "Email cannot be an empty field",
+        "any.required": "Email is required"
+    }),
+    password: Joi.string().min(8).pattern(new RegExp("^[a-zA-Z0-9]{3,30}$")).required().messages({
```
Comment (key-joshua) [inline src/modules/auth/validation/authValidations.ts:34]: Password validation should accept Letters, Numbers and Special Characters. For now it's not accepting special characters.

**Comment by key-joshua** [comment]
Your endpoint should accept form data as user will need to upload profile picture.

**Comment by key-joshua** [comment]
Your endpoint should accept form data as user will need to upload profile picture.

image

File: src/modules/auth/validation/authValidations.ts (hunk 3 above)
Comment (AimePazzo) [reply src/modules/auth/validation/authValidations.ts:34]: Fixed

File: .env.example [hunk 4]
```diff
@@ -1,18 +1,10 @@
 PORT=
```
Comment (key-joshua) [inline .env.example:1]: This env example should get updated with correct variables please

Comment (AimePazzo) [reply .env.example:1]: Done

**Pull request merged**